Release v0.17.0 (#751)
informalsystems/tendermint-rs: CHANGELOG.md

  WebSocketClient for more robust event subscriptions. It can be enabled with the
  `websocket-client` feature. Subscriptions are exposed using unbounded
  channels. ([#516])
- `[tendermint]` Removed all traces of Amino, including `amino_types` modules.

>   All types are now `DomainType`s implementing Protobuf-encoding using Prost.

Reviewer: ```suggestion
  All types are now "domain types" implementing the `Protobuf` trait for Protobuf-encoding using Prost.
```

  ([#504], [#535], [#536], [#585])
- `[tendermint]` Protocol breaking changes for compatibility with Tendermint
  Core v0.34 (and the Cosmos Stargate release) ([#305]):
  - Validators are now sorted by voting power (descending)
    and address (ascending). ([#506])
  - Remove PubKey field from DuplicateVoteEvidence ([#502])
  - Fix hash of empty Merkle tree to comply with RFC6962 ([#498])
  - All binary encoding is done via protobuf3 instead of amino
    ([#504], [#535], [#536], [#585])
  - Various updates to JSON encoding ([#505])
- `[tendermint]` Direct serialization capabilities have been removed from the
  domain types. ([#639])
- `[tendermint]` Work has started on making it compulsory to construct domain
  types by way of their constructors to ensure validity. ([#639])

> - `[tendermint-proto]` The `DomainType` trait has now been renamed to
>   `Protobuf` to clarify its purpose throughout the codebase. ([#672])

Reviewer: ```suggestion
```

Author: Good catch! I wonder how this happened...

  and the `tendermint-rpc` crate's docs for more details ([#516])
- `[rpc]` The subscription client interface provides a structured `Query`
  mechanism to help ensure compile-time validity of subscription queries.
  See the crate docs and [#584] for details.
- `[rpc]` The RPC request and response types' fields are now all publicly
  accessible ([#636]).
- `[rpc]` A new RPC probe (in the `rpc-probe` directory) has been added to
  facilitate quick, pre-scripted interactions with a Tendermint node (via its
  WebSocket endpoint). This aims to help improve testing and compatibility
  between Tendermint in Go and Rust. ([#653])
- `[rpc]` The `WebSocketClient` now adds support for all remaining RPC requests
  by way of implementing the `Client` trait ([#646])
- `[rpc]` Support for the `tx_search` RPC endpoint has been added ([#701])
- `[rpc]` Responses that include events now automatically have their tag
  key/value pairs decoded from base64, where previously tag key/value pairs
  were Base64-encoded ([#717])
- `[rpc]` Support for the `consensus_state` RPC endpoint has been added ([#719])

> - `[tendermint]` Bech32 encoding fix ([#690])

Reviewer: ```suggestion
```
(This bug+fix was within the rc releases. Between 0.16->0.17, this didn't actually change.)

- `[tendermint]` Remove `total_voting_power` parameter from `validator::Set::new` ([#739])
- `[tendermint, rpc, light-client]` Crates now compile to WASM on the
  `wasm32-unknown-unknown` and `wasm32-wasi` targets ([#463])
- `[tendermint, light-client]` Specify the proposer in the validator set of fetched light blocks ([#705])
- `[tendermint-proto]` Upgrade protobuf definitions to Tendermint Go v0.34.0 ([#737])
- `[testgen]` Compute `last_block_id` hash when generating a `LightChain` ([#745])
- Dependency updates:
  - Update sled to 0.34 ([#490])
  - Update k256 to v0.7 ([#752])
  - Remove tai64 crate  ([#603])

### BUG FIXES:

- `[light-client]` Fix bug where a commit with only absent signatures would be
  deemed valid instead of invalid ([#650])
- `[light-client]` Revert a change introduced in [#652] that would enable DoS attacks,
  where full nodes could spam the light client with massive commits (eg. 10k validators).
- `[rpc]` Correctly handles control and keep-alive messages ([#516], [#590])
- `[rpc]` More robust handling of concurrency issues ([#311], [#313])

> - `[tendermint]` CanonicalBlockId is now correctly decoded to `None`
>   in `CanonicalVote` and `CanonicalProposal` when its hash is empty.

Reviewer: ```suggestion
```
I have a feeling this wasn't a problem in 0.16 either.

- `[tendermint]` Time serialization fix (part of [#665])
- `[tendermint]` Fix abci::Data serialization to base64-encoded string. ([#667])
- `[tendermint]` Simplify abci::Log serialization ([#667])
- `[tendermint]` consensus::State backwards compatibility for deserialization ([#679])

> - `[tendermint-proto]` Timestamp serialization fix (part of [#665])

Reviewer: I guess these might have also been opened because of the upgrade, although there might be fixes in there. I'm not sure.

Author: We could easily take them out if you want?

```suggestion
```
